DR-1208 bulk load performance tuning

ddietterich opened the pull request:
Three changes:
1. Improved driver performance by only waving if there is nothing to do.
2. Change application properties to allow for many more stairway threads and many more concurrent file loads
3. Fix a bug in migrate where it was leaving cached connection set to read only

ddietterich committed:
Make in just driver only wait if there's nothing to do.
Changed migrate to use jdbc template.
Increase load file concurrency.

ddietterich committed:
Force new transaction

ddietterich committed:
Revert to manual connections, and take care that they are all closed and not set to read only.

ddietterich committed:
Revert to manual connections, and take care that they are all closed and not set to read only.

ddietterich committed:
Check style

fboulnois commented on src/main/java/bio/terra/service/upgrade/Migrate.java:
```diff
@@ -199,16 +204,20 @@ private void releaseDeploymentLock() {
-                // Application of the migration released the lock for us. Nothing for us to do.
-                return;
-            }
+            try {
```
The triple-nesting of the `try ... catch` logic makes it a bit difficult to reason about this section. Can a level of nesting be avoided here? Return the result of `commitReadOnlyTransaction` and then have something like:

```
} catch (SQLException ex) {
  [ ... ]
} finally {
  if(startedTransaction) {
    commitReadOnlyTransaction(connection);
  }
}
```

ddietterich replied:
Yeah, the problem is that `connection` is out of scope in the finally clause.
I split the inner part of the loop into his separate method and added the comment. See if that helps.

fboulnois reviewed (changes requested):
One suggestion:

RoriCremer commented on src/main/java/bio/terra/service/filedata/flight/ingest/IngestDriverStep.java:
```diff
@@ -143,11 +143,13 @@ public StepResult undoStep(FlightContext context) {
         while (true) {
-            waiting();
+            // Is evaluating the  state of the load candidates is not instant,
+            // we recheck right away before waiting only wait if we did not find any new completions3
```
I'm having a hard time understanding this comment

ddietterich replied:
Grumble! This voice dictation is a real pita

ddietterich committed:
Merge branch 'develop' of github.com:DataBiosphere/jade-data-repo into dd-DR-1208-perf-tune

ddietterich committed:
Refactor overly nested code in migrate
Fix bizarre comment in ingest driver step
Set dropAllOnStart=true were connected tests, so they do not try to
recover flights from a previous run!
fix one case of removed eat database tracking

snf2ye reviewed (approved):
I took a detailed look at this PR earlier this week. It still looks good to me!

ddietterich committed:
a better refactoring of on migrate code

fboulnois reviewed (approved):
latest changes lgtm!
